crates/starknet-server/src/api/http/endpoints/mint_token.rs

```diff
@@ -46,20 +50,30 @@ pub(crate) async fn mint(
     Json(request): Json<MintTokensRequest>,
     Extension(state): Extension<HttpApiHandler>,
 ) -> HttpApiResult<Json<MintTokensResponse>> {
-    // increase balance
     let mut starknet = state.api.starknet.write().await;
+    let mut unit = "WEI".to_string();
+    let mut erc20_contract = ETH_ERC20_CONTRACT_ADDRESS;
+
+    // if unit is FRI, change contract address and unit
+    if let Some(u) = request.unit {
+        if u == "FRI" {
+            erc20_contract = STRK_ERC20_CONTRACT_ADDRESS;
+            unit = u;
+        } else if u != "WEI" {
+            return Err(HttpApiError::MintingError {
+                msg: "Minting is supported only for WEI/FRI unit".to_string(),
+            });
+        }
+    }
```

Comment:
A more idiomatic and concise way would be to use `unwrap_or` and `match`. An even better approach would be to use an enum, preferably an existing one, but here's how it should look without `if`s and `else`s. I also changed variable name from `erc20_contract` to `erc20_address`:

```suggestion
    let unit = request.unit.unwrap_or("WEI".into());

    let erc20_address = match unit.as_str() {
        "WEI" => ETH_ERC20_CONTRACT_ADDRESS,
        "FRI" => STRK_ERC20_CONTRACT_ADDRESS,
        other => {
            return Err(HttpApiError::MintingError {
                msg: format!("Minting is supported only for WEI/FRI, got: {other}"),
            })
        }
    };
```